library/src/G4CMPUtils.cc

```diff
@@ -228,40 +228,30 @@ void G4CMP::FillHit(const G4Step* step, G4CMPElectrodeHit* hit) {
 
 // Generate cos(theta) law for diffuse reflection, ensuring that computed
 // vector is directed inward with respect to the surface normal.
-
+// IMPORTANT: We note here that this function assumes a "generalized"
+// surface norm, i.e. a norm that is opposite the incoming vel/mom
```

Comment:
Technically, it's more complicated than that -- for interior reflection, the surface norm is in the same direction as the incoming velocity.  For reflection off of an exterior volume, the surface norm would be opposite the inbound direction.

Reply:
Oh!  It sounds like I had the wrong picture in my head.  So you've "changed" the SurfaceNorm convention globally so that it is now everywhere the "inward pointing" normal, rather than the outward-pointing normal that Geant4 uses.  

I thought that what "genrealized" was doing was taking into account the complicated mother-daughter or daughter-mother boundaries, but keeping the direction of the normal as it "should be."  But you and I had different visions of what "should be" actually meant.

I'm still concerned (and even more so, now) that you've adopted the opposite convention to Geant4 itself.  I think that should be resolved, but this comment thread isn't the place to discuss it.